src/main/pages/che-7/hands-on-guidance/proc_creating-a-workspace-by-importing-source-code-of-your-project.adoc

```diff
@@ -1,27 +1,99 @@
 [id="creating-a-workspace-by-importing-source-code-of-your-project_{context}"]
 = Creating a workspace by importing source code of your project
 
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section is about how to create a new workspace to edit an existing codebase. There are 3 ways to do that:
+
+* xref:creating-a-workspace-from-code-sample_{context}#change-the-workspace-configuration[Select a stack from the User Dashboard but change the project]
+* Add a devfile to a git repository and starting the workspace using chectl or a factory
+* Import a project in a running workspace from the editor
 
 [discrete]
 == Prerequisites
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
+* You need an OKD, OCP, or OSD cluster, if using Che on Openshift. See below for use of link:https://www.okd.io/minishift/[Minishift].
+* Or, you need a Kubernetes cluster such as link:https://github.com/kubernetes/minikube#installation[Minikube] to use Che on Kubernetes.
 
 [discrete]
-== Procedure
+== Importing from the dashboard
```

Comment:
You talk about adding a project to an already existing workspace. Not about creating a new workspace using a stack and adding an existing project (i.e. overriding the default sample that comes with the stack). That's an interesting use case as well but this section is about "Creating a workspace...".

Reply:
ah, so the workflow here should be:
* pick a stack from the list
* Create & Proceed Editing (which aside, is bad engrish)
* configure devfile to pull a different default project
* open workspace

Reply:
well you don't need to `Proceed Editing`, you can:
- select a stack
- select `git` or `github` tab and fill the details of the repo you want to add as a project
- click `Add project`
- Create & Open Workspace

Reply:
ok, cool I have that one documented already in http://0.0.0.0:4000/che-7/end-user-tasks.html#edit-an-existing-repository (inside http://0.0.0.0:4000/che-7/end-user-tasks.html#importing-from-the-user-dashboard-into-an-existing-workspace ) 

![image](https://user-images.githubusercontent.com/227597/60530606-1b2efd00-9cc7-11e9-93b6-045c0972ce6b.png)